src/web/app/components/feedback-path-panel/feedback-path-panel.component.spec.ts

```diff
@@ -26,10 +29,89 @@ describe('FeedbackPathPanelComponent', () => {
+    ]);
+    component.triggerModelChangeBatch = new EventEmitter<any>();
+    emitSpy = jest.spyOn(component.triggerModelChangeBatch, 'emit');
     fixture.detectChanges();
   });
 
   it('should create', () => {
     expect(component).toBeTruthy();
   });
+
+  it('should toggle sub menu status correctly', () => {
+    const menu = FeedbackParticipantType.STUDENTS;
+    component.toggleSubMenu(menu);
+    const firstToggle = component.isSubMenuOpen(menu);
+    component.toggleSubMenu(menu);
+    const secondToggle = component.isSubMenuOpen(menu);
+    expect(firstToggle).toBe(!secondToggle);
+  });
+
+  it('should reset menu', () => {
+    const menu = FeedbackParticipantType.STUDENTS;
+    component.toggleSubMenu(menu);
+    component.resetMenu();
+    const isMenuOpen = component.isSubMenuOpen(menu);
+    expect(isMenuOpen).toBe(false);
+  });
+
+  it('should trigger custom number of entities', () => {
+    const customNumber = 5;
+    const customNumberOfEntitiesToGiveFeedbackToEmitSpy =
+      jest.spyOn(component.customNumberOfEntitiesToGiveFeedbackTo, 'emit');
+    component.triggerCustomNumberOfEntities(customNumber);
+    expect(customNumberOfEntitiesToGiveFeedbackToEmitSpy).toHaveBeenCalledWith(customNumber);
+  });
+
+  it('should trigger number of entities setting', () => {
+    const setting = NumberOfEntitiesToGiveFeedbackToSetting.CUSTOM;
+    const numberOfEntitiesToGiveFeedbackToSettingEmitSpy =
+      jest.spyOn(component.numberOfEntitiesToGiveFeedbackToSetting, 'emit');
+    component.triggerNumberOfEntitiesSetting(setting);
+    expect(numberOfEntitiesToGiveFeedbackToSettingEmitSpy).toHaveBeenCalledWith(setting);
+  });
+
+  it('should trigger custom feedback path', () => {
+    const customFeedbackPathEmitSpy = jest.spyOn(component.customFeedbackPath, 'emit');
+    component.triggerCustomFeedbackPath();
+    expect(customFeedbackPathEmitSpy).toHaveBeenCalledWith(true);
+  });
+
+  describe('changeGiverRecipientType', () => {
+    it('should set default recipientType if recipientType is not allowed for giverType', () => {
+      component.allowedFeedbackPaths.set(FeedbackParticipantType.TEAMS,
+        [FeedbackParticipantType.OWN_TEAM_MEMBERS, FeedbackParticipantType.STUDENTS]);
+      component.changeGiverRecipientType(FeedbackParticipantType.TEAMS, FeedbackParticipantType.INSTRUCTORS);
+      expect(component.model.recipientType).toEqual(FeedbackParticipantType.OWN_TEAM_MEMBERS);
+    });
+
+    it('checks if custom path remains & emits false if input unchanged', () => {
+      component.model.isUsingOtherFeedbackPath = true;
+      component.model.giverType = FeedbackParticipantType.TEAMS;
+      component.model.recipientType = FeedbackParticipantType.STUDENTS;
+      component.changeGiverRecipientType(FeedbackParticipantType.TEAMS, FeedbackParticipantType.STUDENTS);
+      expect(emitSpy).toHaveBeenCalledWith({
+        isUsingOtherFeedbackPath: false,
+      });
+    });
```

Comment:
small nit here, let's add another test to check for `isUsingOtherFeedbackPath = false` as well and ensure that emitSpy is not called